### PR TITLE
Add NI-FGEN proto file and update generated `nifgen` files.

### PR DESCRIPTION
**Pull request by tdunkle**
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).

~- [ ] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~

This change does not need to be customer-facing.

~- [ ] I've added tests applicable for this pull request~

Showing that the existing NI-FGEN system-level tests pass is sufficient.

### What does this Pull Request accomplish?

This pull request adds the NI-FGEN proto file to nimi-python, so that the NI-FGEN python API has a gRPC option enabled.
Note: This PR is a follow-on to #1830, which was accidentally closed

### List issues fixed by this Pull Request below, if any.

* [Internal NI bug AB#2160599](https://ni.visualstudio.com/DevCentral/_workitems/edit/2160599/)

### What testing has been done?

Installed python 3.7.9 and synced the nimi-python repo to a VM that has NI-FGEN 21.8 installed. Observed that I could build fgen with `tox -e py37-codegen -- nifgen`, and that I could run the system level tests with `tox -c generated\nifgen\tox-system_tests.ini`, and they all passed. Test output, for reference: [fgen_test_output.txt](https://github.com/ni/nimi-python/files/9778276/fgen_test_output.txt)

### CODE REVIEW

File: src/nifgen/metadata/nifgen.proto [hunk 1]
```diff
@@ -0,0 +1,2150 @@
+
+//---------------------------------------------------------------------
+// This file is generated from NI-FGEN API metadata version 23.0.0
```
Comment (marcoskirsch) [inline src/nifgen/metadata/nifgen.proto:3]: New exports contain the build number as well, so this is a bit outdated.
Can you check latest?

Comment (marcoskirsch) [reply src/nifgen/metadata/nifgen.proto:3]: example: https://p4.natinst.com/fsm/perforce/build/exports/ni/nifg/nifgen_grpc_device/official/export/23.0/.latestFinal/metadata/

Comment (tdunkle) [reply src/nifgen/metadata/nifgen.proto]: Updated to d41 proto file.

File: generated/nifgen/nifgen/_grpc_stub_interpreter.py [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+        )
+
+    def is_done(self):  # noqa: N802
+        response = self._invoke(
+            self._client.IsDone,
+            grpc_types.IsDoneRequest(vi=self._vi),
+        )
+        return response.done
+
+    def lock(self):  # noqa: N802
```
Comment (dmondrik) [inline generated/nifgen/nifgen/_grpc_stub_interpreter.py:352]: Tom: You need to bring in the latest metadata exports from `nifgen_nimi_python` as well, into your metadata folder. That contains a change that will alter what gets generated for the lock method here.

Comment (tdunkle) [reply generated/nifgen/nifgen/_grpc_stub_interpreter.py:352]: Done!

Comment (dmondrik) [reply generated/nifgen/nifgen/_grpc_stub_interpreter.py:352]: Actually Tom you need to re-run `tox` and push again. This code should look different.

**Review by marcoskirsch — APPROVED** [review]

**Pull request merged**